pkg/microservice/systemconfig/core/codehost/service/codehost.go

```diff
@@ -132,6 +138,18 @@ func UpdateCodeHost(host *models.CodeHost, _ *zap.SugaredLogger) (*models.CodeHo
 	if host.Type == setting.SourceFromGerrit {
 		host.AccessToken = base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%s:%s", host.Username, host.Password)))
 	}
+
+	var oldAlias string
+	oldCodeHost, err := mongodb.NewCodehostColl().GetCodeHostByID(host.ID)
+	if err == nil {
+		oldAlias = oldCodeHost.Alias
+	}
+	if host.Alias != "" && host.Alias != oldAlias {
+		if _, err := mongodb.NewCodehostColl().GetCodeHostByAlias(host.Alias); err == nil {
+			return nil, fmt.Errorf("alias cannot have the same name")
```

Comment:
what about the old data, like same repo integrated twice?

Reply:
When modifying, compare it with the current value. If it is the same as the current value, do not check the duplicate.